test/integration/objectstore/test_jobs.py

```diff
@@ -79,17 +98,39 @@ def test_tool_simple_constructs(self):
         assert files_3_count == 0
 
         # Ensure the 10 inputs were written to one of the distributed object store's disk
-        # stores (it will have either 10 or 11 depeending on whether the input was also
+        # stores (it will have either 10 or 11 depending on whether the input was also
         # written there. The other disk store may or may not have the input file so should
         # have at most one file.
-        assert (files_1_count >= 10) or (files_2_count >= 10)
-        assert (files_1_count <= 1) or (files_2_count <= 1)
+        assert (files_1_count + files_2_count == 10) or (files_1_count + files_2_count == 11)
 
         # Other sanity checks on the test - just make sure the test was setup as intended
         # and not actually testing object store behavior.
         assert (files_1_count <= 11) and (files_2_count <= 11)
         assert (files_1_count >= 0) and (files_2_count >= 0)
 
+        # TODO: ideally the following assertion should be separated in a different test method.
+        contents = []
+        path1_files = _get_datasets_files_in_path(self.files1_path)
+        path2_files = _get_datasets_files_in_path(self.files2_path)
+        path3_files = _get_datasets_files_in_path(self.files3_path)
+        for filename in set().union(path1_files, path2_files, path3_files):
+            with open(filename) as f:
+                content = f.read().strip('\n').strip('\t')
```

Comment:
```suggestion
                content = f.read().strip()
```
I don't see where the tab should be coming from.

Reply:
no tab, but better to strip the line break to simplify content assertions.

Reply:
line break (and tabs for that matter) are considered whitespace, so they are removed by `strip()`:
```
In [1]: s = ''

In [2]: ?s.strip
Docstring:
S.strip([chars]) -> str

Return a copy of the string S with leading and trailing
whitespace removed.
If chars is given and not None, remove characters in chars instead.
Type:      builtin_function_or_method

In [3]: s = '\n\t'

In [4]: s.strip()
Out[4]: ''
```

Reply:
Good point, thanks.